addon/adapters/application.js

```diff
@@ -1,66 +1,140 @@
+var get = Ember.get,
+    forEach = Ember.ArrayPolyfills.forEach;
 
-  defaultSerializer: '-parse',
 
-  init: function(){
+export default DS.RESTAdapter.extend({
+  host: 'https://api.parse.com',
+  namespace: '1',
+  classesPath: 'classes',
+  parseClientVersion: 'js1.4.2',
+
+  init() {
     this._super();
 
-    this.set( 'headers', {
-      'X-Parse-Application-Id' : Ember.get( this, 'applicationId' ),
-      'X-Parse-REST-API-Key'   : Ember.get( this, 'restApiId' )
-    });
+    this.set('installationId', this._getInstallationId());
+    this.set('sessionToken', null);
+    this.set('userId', null);
+
+    /*
+     * avoid pre-flight.
+     * Parse._ajax
+     */
+    this.set('headers', { 'Content-Type': 'text/plain' });
   },
 
-  host: 'https://api.parse.com',
+  _getInstallationId() {
+    /*
+     * Parse._getInstallationId
+     */
+    var hexOctet = function() {
+      return (
+        Math.floor((1+Math.random())*0x10000).toString(16).substring(1)
+      );
+    };
 
-  namespace: '1',
+    return (
+      hexOctet() + hexOctet() + "-" +
+      hexOctet() + "-" +
+      hexOctet() + "-" +
+      hexOctet() + "-" +
+      hexOctet() + hexOctet() + hexOctet());
+  },
 
-  classesPath: 'classes',
+  ajaxOptions(url, type, options) {
+    var hash = options || {};
+    hash.data = hash.data || {};
+    hash.url = url;
+    hash.type = type;
+    hash.dataType = 'json';
+    hash.context = this;
+
+    if ((hash.data && type !== 'GET')) {
+      hash.contentType = 'application/json; charset=utf-8';
+
+      // Parse auth stuff
+      hash.data._ClientVersion = this.get('parseClientVersion');
+      hash.data._ApplicationId = this.get('applicationId');
+      hash.data._JavaScriptKey = this.get('javascriptKey');
+      hash.data._InstallationId = this.get('installationId');
+
+      var _sessionToken = this.get('sessionToken');
+      if (_sessionToken) {
+        hash.data._SessionToken = _sessionToken;
+      }
+
+      hash.data = JSON.stringify(hash.data);
+    }
+
+    var headers = get(this, 'headers');
+    if (headers !== undefined) {
+      hash.beforeSend = function (xhr) {
+        forEach.call(Ember.keys(headers), function(key) {
+          xhr.setRequestHeader(key, headers[key]);
+        });
+      };
+    }
+
+    return hash;
+  },
+
+  ajaxError(jqXHR, responseText, errorThrown) {
+    if (jqXHR.responseJSON.error === 'invalid session token') {
+      // invalid session
+      var session = this.container.lookup('service:session');
+      session.resetSession();
+    }
 
-  pathForType: function( type ) {
-    if ( 'parseUser' === type ) {
+    return this._super(jqXHR, responseText, errorThrown);
+  },
+
+  normalizeErrorResponse: function(status, headers, payload) {
+    return [
+      {
+        status: `${status}`,
+        title: 'The backend responded with an error',
+        details: payload.error,
+        code: payload.code
+      }
+    ];
+  },
+
+  pathForType(type) {
```

Comment:
It is a dead horse :-)

I don't think there is conceptual disagreement. The challenge is that the adapter needs to know who the current user is for access to certain data. The integration point needs to be designed, and it seems folly to just rip the code out without having documented and implemented the replacement path.